packages/react-dom/src/__tests__/ReactComponentLifeCycle-test.js

```diff
@@ -830,4 +830,49 @@ describe('ReactComponentLifeCycle', () => {
       'UNSAFE_componentWillUpdate',
     ]);
   });
+
+  it('should not override state with stale values if prevState is spread within getDerivedStateFromProps', () => {
+    let childInstance;
+
+    class Child extends React.Component {
+      state = {local: 0};
+      static getDerivedStateFromProps(nextProps, prevState) {
+        return {...prevState, remote: nextProps.remote};
+      }
+      updateState = () => {
+        this.setState(state => ({local: state.local + 1}));
+        this.props.onChange(this.state.remote + 1);
+      };
+      render() {
+        childInstance = this;
+        return (
+          <div onClick={this.updateState}>{`remote:${
+            this.state.remote
+          }, local:${this.state.local}`}</div>
+        );
+      }
+    }
+
+    class Parent extends React.Component {
+      state = {value: 0};
+      handleChange = value => {
+        this.setState({value});
+      };
+      render() {
+        return <Child remote={this.state.value} onChange={this.handleChange} />;
+      }
+    }
+
+    const instance = ReactTestUtils.renderIntoDocument(<Parent />);
+    const element = ReactDOM.findDOMNode(instance);
```

Comment:
Nit: let's not use `findDOMNode`. Maybe a ref instead? I know it's just a test, but we should still try to avoid using legacy APIs :D